diff -up quesoglc-0.7.2/src/scalable.c~ quesoglc-0.7.2/src/scalable.c
--- quesoglc-0.7.2/src/scalable.c~	2008-06-04 23:47:51.000000000 +0200
+++ quesoglc-0.7.2/src/scalable.c	2024-01-20 16:31:45.595265108 +0100
@@ -596,16 +596,10 @@ void __glcRenderCharScalable(__GLCfont*
     gluTessProperty(tess, GLU_TESS_WINDING_RULE, GLU_TESS_WINDING_ODD);
     gluTessProperty(tess, GLU_TESS_BOUNDARY_ONLY, GL_FALSE);
 
-    gluTessCallback(tess, GLU_TESS_ERROR,
-			(void (CALLBACK *) (GLenum))__glcCallbackError);
-    gluTessCallback(tess, GLU_TESS_VERTEX_DATA,
-		    (void (CALLBACK *) (void*, void*))__glcVertexCallback);
-    gluTessCallback(tess, GLU_TESS_COMBINE_DATA,
-		    (void (CALLBACK *) (GLdouble[3], void*[4],
-					GLfloat[4], void**, void*))
-		    __glcCombineCallback);
-    gluTessCallback(tess, GLU_TESS_BEGIN_DATA,
-		    (void (CALLBACK *) (GLenum, void*))__glcBeginCallback);
+    gluTessCallback(tess, GLU_TESS_ERROR, (_GLUfuncptr)__glcCallbackError);
+    gluTessCallback(tess, GLU_TESS_VERTEX_DATA, (_GLUfuncptr)__glcVertexCallback);
+    gluTessCallback(tess, GLU_TESS_COMBINE_DATA, (_GLUfuncptr)__glcCombineCallback);
+    gluTessCallback(tess, GLU_TESS_BEGIN_DATA, (_GLUfuncptr)__glcBeginCallback);
 
     gluTessNormal(tess, 0., 0., 1.);
 
diff -up quesoglc-0.7.2/src/unicode.c~ quesoglc-0.7.2/src/unicode.c
--- quesoglc-0.7.2/src/unicode.c~	2024-01-20 16:35:43.000000000 +0100
+++ quesoglc-0.7.2/src/unicode.c	2024-01-20 16:41:27.195275693 +0100
@@ -550,7 +550,7 @@ GLCchar32* __glcConvertToVisualUcs4(__GL
 {
   GLCchar32* string = NULL;
   int length = 0;
-  FriBidiCharType base = FRIBIDI_TYPE_ON;
+  FriBidiParType base = FRIBIDI_TYPE_ON;
   GLCchar32* visualString = NULL;
 
   assert(inString);
@@ -677,7 +677,7 @@ GLCchar32* __glcConvertCountedStringToVi
 						const GLint inCount)
 {
   GLCchar32* string = NULL;
-  FriBidiCharType base = FRIBIDI_TYPE_ON;
+  FriBidiParType base = FRIBIDI_TYPE_ON;
   GLCchar32* visualString = NULL;
 
   assert(inString);
