--- a/src/sleep/sleep.c	2025-01-13 16:07:01.000000000 +0100
+++ b/src/sleep/sleep.c	2025-07-08 01:01:13.941946081 +0200
@@ -676,7 +676,7 @@
                 if (r < 0)
                         return r;
 
-                r = fd_wait_for_event(tfd, POLLIN, 0);
+                r = fd_wait_for_event(tfd, POLLIN, 500*1000);
                 if (r < 0)
                         return log_error_errno(r, "Error polling timerfd: %m");
                 /* Store fd_wait status */
